scripts/data_files/driver_jsons/mbedtls_test_opaque_driver.json

```diff
@@ -0,0 +1,20 @@
+{
```

Comment:
Let me answer this with how the driver JSONS were intended to be placed.
The generate_driver_wrappers.py script takes a parameter to point to the location of the driver JSONS. But by default points to a new directory created under scripts => scripts/data_files/driver_jsons/.

Do we want to make the test file path the default path ? and keep it under tests/driver_jsons ? What is the additional nesting you speak of ? 

If we have the default directory at a common path like scripts/data_files/driver_jsons/ there is an advantage of adding in user driver jsons as they grow and editing the file driverlist.json to create the psa_crypto_driver_wrapper.c file only with the files listed in driverlist.json. So driverlist.json is the final gatekeeper.

I think this is far simpler than having to pass it as an argument every time, while still giving the user the option to pass a different location if needed.

Reply:
Eventually, no test-specific code or configuration should be outside `tests`. We initially added test-specific code in `library/*.c` because we didn't have the infrastructure to do it another way, and you're now adding the required infrastructure. We're still in a transitional period, and it's ok if the test driver description files are initially placed under `scripts/data_files/` and move to `tests/` in a follow-up. But I think it's easier to put them in their final place from the start.

Eventually, a default build of Mbed TLS should either not include any driver dispatch, or driver dispatch for accelerators with some conditional inclusion based on symbols from `mbedtls_config.h` and `psa/crypto_config.h`. If there is some support for drivers, this should still not include test driver code without a non-default build option. It's too early to make that switch, though: for now the default build should include the test drivers if the JSON files are unchanged and `MBEDTLS_PSA_CRYPTO_DRIVERS` is enabled. We need to convert all the test driver dispatch to templates first.

So for now I'd expect the default driver list to point to the test driver description files. But I don't see why the test driver description files themselves shouldn't be under `tests/`.

Reply:
The reason I wouldn't want them under tests/ is only because of scripting annoyance, then the default option only points to the test drivers and we will need to pass in a different location for any other case. I would prefer the default driver to be a place where users can add in their driver.jsons while fully able to overload it with a different location if need be.

Any edit to the driverList.json ensures what actual drivers are used for autogen and I am assuming we've already agreed on that !

Reply:
Eventually there shouldn't be any references to `tests` from `library` or `scripts`. But we're not there yet, so ok, let's keep the test driver templates where they are for now, and we'll move them later as part of removing the temporary references to test drivers from library code.